addons/account/models/account_move.py

```diff
@@ -1202,11 +1202,28 @@ def _compute_tax_totals(self):
                             handle_price_include=False,
                         ))
                 move.tax_totals = self.env['account.tax']._prepare_tax_totals(**kwargs)
-                rounding_line = move.line_ids.filtered(lambda l: l.display_type == 'rounding')
-                if rounding_line:
-                    amount_total_rounded = move.tax_totals['amount_total'] + sign * rounding_line.amount_currency
-                    move.tax_totals['amount_total_rounded'] = amount_total_rounded
-                    move.tax_totals['formatted_amount_total_rounded'] = formatLang(self.env, amount_total_rounded, currency_obj=move.currency_id) or ''
+                if move.invoice_cash_rounding_id:
+                    rounding_amount = move.invoice_cash_rounding_id.compute_difference(move.currency_id, move.tax_totals['amount_total'])
+                    totals = move.tax_totals
+                    totals['display_rounding'] = True
+                    if rounding_amount:
+                        if move.invoice_cash_rounding_id.strategy == 'add_invoice_line':
+                            totals['rounding_amount'] = rounding_amount
+                            totals['formatted_rounding_amount'] = formatLang(self.env, totals['rounding_amount'], currency_obj=move.currency_id)
+                            totals['amount_total_rounded'] = totals['amount_total'] + rounding_amount
+                            totals['formatted_amount_total_rounded'] = formatLang(self.env, totals['amount_total_rounded'], currency_obj=move.currency_id)
+                        elif move.invoice_cash_rounding_id.strategy == 'biggest_tax':
+                            if totals['subtotals_order']:
+                                subtotal_with_biggest_tax_amount, tax_group_index, _ = max((
+                                    (subtotal, tax_group_index, tax_group['tax_group_amount'])
+                                    for subtotal, tax_groups in totals['groups_by_subtotal'].items()
+                                    for tax_group_index, tax_group in enumerate(tax_groups)
+                                ), key=lambda x: x[2])
+                                biggest_tax = totals['groups_by_subtotal'][subtotal_with_biggest_tax_amount][tax_group_index]
```

Comment:
Isn't this equivalent? (not tested)
```suggestion
                                biggest_tax = max((
                                    tax_group['tax_group_amount']
                                    for tax_groups in totals['groups_by_subtotal'].values()
                                    for tax_group in tax_groups
                                ), key=lambda tax_group: tax_group['tax_group_amount'])
```

Reply:
Indeed much cleaner, I just had to change the first `tax_group['tax_group_amount']` into `tax_group`